python/lsst/meas/pz/estimate_pz_task.py

```diff
@@ -0,0 +1,481 @@
+
+
+class EstimatePZConnections(
+    PipelineTaskConnections, dimensions=("instrument", "tract", "patch")
+):
+    """Connections for tasks that make p(z) estimates
+
+    These will take pickled model file as a "calibration-like" input,
+    an objectTable as input, and create a p(z) file in 'qp' format.
+    """
+
+    pzModel = cT.PrerequisiteInput(
+        doc="Model for PZ Estimation",
+        name="pzModel",
+        storageClass="PZModel",
+        dimensions=["instrument"],
+        isCalibration=True,
+        # lookupFunction=_pzModelLookup,
+    )
+
+    objectTable = cT.Input(
+        doc="Object table in parquet format, per patch",
+        name="objectTable",
+        storageClass="DataFrame",
+        dimensions=(
+            "skymap",
+            "tract",
+            "patch",
+        ),
+        deferLoad=True,
+    )
+
+    pzEnsemble = cT.Output(
+        doc="Per-object p(z) estimates, per patch",
+        name="pzEnsemble",
+        storageClass="QPEnsemble",
+        dimensions=(
+            "skymap",
+            "tract",
+            "patch",
+        ),
+    )
+
+
+class EstimatePZAlgoConfigBase(
+    pexConfig.Config,
+):
+    """Base class for configurations of algorithm specific p(z)
+    estimation tasks.
+
+    This class mostly just translates the RAIL configuration
+    parameters to pex.config parameters.
+
+    the `_make_fields` method will import the RAIL estimation stage
+    and loop through the stage config parameters and make corresponding
+    pex.config parameters.
+
+    Subclasses will just have to set
+    `stage_class` and invoke _make_fields.
+    """
+
+    stage_class = None
+
+    stage_name = pexConfig.Field(doc="Rail stage name", dtype=str)
+    mag_offset = pexConfig.Field(doc="Magnitude offset", dtype=float, default=31.4)
+    flux_column_template = pexConfig.Field(
+        doc="Template for flux column names",
+        dtype=str,
+        default="{band}_gaap1p0Flux",
+    )
+    flux_err_column_template = pexConfig.Field(
+        doc="Template for flux error column names",
+        dtype=str,
+        default="{band}_gaap1p0FluxErr",
+    )
+    mag_template = pexConfig.Field(
+        doc="Template for magnitude names",
+        dtype=str,
+        default="mag_{band}_lsst",
+    )
+    mag_err_template = pexConfig.Field(
+        doc="Template for magntitude error names",
+        dtype=str,
+        default="mag_err_{band}_lsst",
+    )
+    band_a_env = pexConfig.DictField(
+        doc="Reddening parameters",
+        keytype=str,
+        default=dict(
+            u=4.81,
+            g=3.64,
+            r=2.70,
+            i=2.06,
+            z=1.58,
+            y=1.31,
+        ),
+    )
+
+    @classmethod
+    def _make_fields(cls):
+        # stage_class = CeciPipelineStage.get_stage(
+        #    cls.estimator_class, cls.estimator_module
+        # )
+        stage_class = cls.estimator_class
+        for key, val in stage_class.config_options.items():
```

Comment:
Yes.  I think that getting the param from rail is better than duplicating them.  I guess if @TallJimbo could give some input on the constraints as to what happens when we update config classes that would help understand how we manage this.

Reply:
I think that the model will have to be that the whoever is _curating_ the photo-z selection algorithms provides the correct settings / parameters, i.e., they basically manage the pipeline files.   If it is helpful to add some doc pointing to rail so that people can find the underlying code / parameters / behavior, that is pretty easy to do.

Reply:
I think disruption from changing config definitions will be tolerable here; I think the worst-case scenario is that we'd have to render some old configs in old processing runs unreadable.

Once we settle on a production photo-z algorithm I think it may be worthwhile to duplicate the configuration for that one out explicitly, but we can live with this in the meantime.